torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -509,28 +509,11 @@ void TracingState::setValue(const IValue& v, Value* value) {
     auto dict = v.toGenericDict();
     TypePtr key_type = dict.keyType();
     TypePtr value_type = dict.valueType();
-    auto dict_size = dict.size();
-    auto handle_unpack = [&](Symbol opname) {
-      auto unpack_to_list = graph->insert(opname, {value});
-      auto list_unpack = graph->createListUnpack(unpack_to_list, dict_size);
-      auto unpack_node = graph->insertNode(list_unpack);
-      auto elem_values = unpack_node->outputs();
-      AT_ASSERT(dict.size() == elem_values.size());
-      size_t i = 0;
-      for (const auto& entry : dict) {
-        if (opname == aten::keys) {
-          setValue(entry.key(), elem_values[i]);
-        } else {
-          setValue(entry.value(), elem_values[i]);
-        }
-        ++i;
-      }
-    };
-    if (key_type->cast<TensorType>()) {
-      handle_unpack(aten::keys);
-    }
-    if (value_type->cast<TensorType>()) {
-      handle_unpack(aten::values);
+    for (const auto& entry : dict) {
```

Comment:
This will actually implies that if dict[tensor, tensor] (tensor as keys) as input or cross tracing/scripting, the tensor key will always be treated as constant, which is ok, but I am wondering if we should throw some warning here to user?

Reply:
Agreed, I filed https://github.com/pytorch/pytorch/issues/40886 as a followup to improve the error messaging in all cases where a key mismatch occurs